yt_dlp/extractor/zdf.py

```diff
@@ -316,12 +357,28 @@ def _extract_entry(self, url, player, content, video_id):
             'timestamp': unified_timestamp(content.get('editorialDate')),
             'thumbnails': thumbnails,
             'chapters': chapters or None,
+            'episode': title,
+            **traverse_obj(content, ('programmeItem', 0, 'http://zdf.de/rels/target', {
+                'series_id': ('http://zdf.de/rels/cmdm/series', 'seriesUuid'),
+                'series': ('http://zdf.de/rels/cmdm/series', 'seriesTitle'),
+                'season': ('http://zdf.de/rels/cmdm/season', 'seasonTitle'),
+                'season_number': ('http://zdf.de/rels/cmdm/season', 'seasonNumber'),
+                'season_id': ('http://zdf.de/rels/cmdm/season', 'seasonUuid'),
+                'episode_number': 'episodeNumber',
+                'episode_id': 'contentId',
```

Comment:
We should type check/transform
```suggestion
                'series_id': ('http://zdf.de/rels/cmdm/series', 'seriesUuid', {str}),
                'series': ('http://zdf.de/rels/cmdm/series', 'seriesTitle', {str}),
                'season': ('http://zdf.de/rels/cmdm/season', 'seasonTitle', {str}),
                'season_number': ('http://zdf.de/rels/cmdm/season', 'seasonNumber', {int_or_none}),
                'season_id': ('http://zdf.de/rels/cmdm/season', 'seasonUuid', {str}),
                'episode_number': ('episodeNumber', {int_or_none}),
                'episode_id': ('contentId', {str}),
```

Reply:
the str typecasts seem to be breaking the tests:

E   AssertionError: '1d7a1879-01ee-4468-8237-c6b4ecd633c7' != None : Invalid value for field series_id, expected '1d7a1879-01ee-4468-8237-c6b4ecd633c7', got None

Reply:
weird. can you send a sample of the API response JSON?

Reply:
actually, no it is using the old extractor now, let me take another look at the other change

Reply:
i forgot a `break` my bad